src/brsTypes/components/RoDeviceInfo.ts

```diff
@@ -0,0 +1,687 @@
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("Default");
+        },
+    });
+
+    private getClockFormat = new Callable("getClockFormat", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("12h");
+        },
+    });
+
+    private enableAppFocusEvent = new Callable("enableAppFocusEvent", {
+        signature: {
+            args: [new StdlibArgument("enable", ValueKind.Boolean)],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("Event notifications are enabled");
+        },
+    });
+
+    private enableScreensaverExitedEvent = new Callable("enableScreensaverExitedEvent", {
+        signature: {
+            args: [new StdlibArgument("enable", ValueKind.Boolean)],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("Screen saver exit event notifications are enabled");
+        },
+    });
+
+    private enableLowGeneralMemoryEvent = new Callable("enableLowGeneralMemoryEvent", {
+        signature: {
+            args: [new StdlibArgument("enable", ValueKind.Boolean)],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("lowGeneralMemoryLevel event notifications are enabled");
+        },
+    });
+
+    private getGeneralMemoryLevel = new Callable("getGeneralMemoryLevel", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("normal");
+        },
+    });
+
+    private isStoreDemoMode = new Callable("isStoreDemoMode", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getLinkStatus = new Callable("getLinkStatus", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private enableLinkStatusEvent = new Callable("enableLinkStatusEvent", {
+        signature: {
+            args: [new StdlibArgument("enable", ValueKind.Boolean)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getConnectionType = new Callable("getConnectionType", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("WiFi Connection");
+        },
+    });
+
+    private getExternalIp = new Callable("getExternalIp", {
```

Comment:
Hmm these IP functions are interesting. Technically, we could get the user's IP and return that. I'm not sure whether we would want to do that though. I'd say let's return blank strings/arrays for now. If we decide we want it in the future, we could always add dynamic IP later.